### DISCUSSION TITLE
Testing setup?

### DISCUSSION THREAD

**Discussion by josmithua** [nandorojo/solito#63] [Q&A]
Hi there,

Is there a guide or recommendations for setting up tests with jest and jest-expo?

I have tests for components and modules in `packages/app`.

Where should I add the test dependencies (jest, jest-expo, @testing-library/{react,jest}-native)?

**Comment by kennym** [comment]
I'm guessing they would go into `packages/app`.

**Comment by nandorojo** [comment]
native dependencies should be in `apps/expo`. maybe take a look at showtime’s repo for some info?

**Comment by Scrump31** [comment]
Since this discussion started, are there any working examples available for setting up @testing-library/react-native or @testing-library/react-native with Jest and Typescript? My team recently decided to use Solito on a project.

**Comment by nandorojo** [comment]
hey guys, Showtime is a real-world Solito app using Jest and Storybook. Maybe reference their repo?

https://github.com/showtime-xyz/showtime-frontend

**Reply by Scrump31** [reply]
@nandorojo Thanks, I will check it out!

**Comment by D-Pagey** [comment]
I've been looking into trying to set up RNTL and got a bit stuck. Wasn't sure if I should be writing tests in `packages/app`, `next` or `expo`. Regardless, when I tried in each of them, ran into issues with TS and config issues. 

In the end, just decided to write E2E Cypress tests in `next` and probably going to look at Detox for the `expo` part. 

Would be nice to be able to write a RNTL integration test in `packages/app`
